packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,485 @@
+ * - Message
+ *   This will prompt the user and require a response in the form of a message. This can be helpful if you require a user to upload an image for example, or give text input.
+ *
+ * You must either use this class directly or extend it.
+ *
+ * [[MessagePrompter]] uses reactions to prompt for a yes/no answer and returns it.
+ * You can modify the confirm and cancel reaction used for each message, or use the [[MessagePrompter.defaultPrompts]].
+ * [[MessagePrompter.defaultPrompts]] is also static so you can modify these directly.
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you sure you want to continue?');
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Choose a number between 5 and 10?', 'number', {
+ * 		start: 5,
+ * 		end: 10
+ * });
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you happy or sad?', 'reaction', {
+ * 		reactions: ['🙂', '🙁']
+ * });
+ * const result = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Do you love me?', 'message');
+ * const result = await handler.run(channel, author);
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The strategy used in [[MessagePrompter.run]]
+	 */
+	public strategy: MessagePrompterBaseStrategy;
+
+	/**
+	 * Constructor for the [[MessagePrompter]] class
+	 * @param message The message to send.
+	 * @param strategy The strategy name or Instance to use
+	 * @param strategyOptions The options that are passed to the strategy
+	 */
+	public constructor(message: MessagePrompterMessage | MessagePrompterBaseStrategy, strategy?: string, strategyOptions?: Record<string, unknown>) {
+		let strategyToRun: MessagePrompterBaseStrategy | undefined = undefined;
+
+		if (message instanceof MessagePrompterBaseStrategy) {
+			strategyToRun = message as MessagePrompterBaseStrategy;
+		} else {
+			const mapStrategy = MessagePrompter.strategies.get(strategy ?? MessagePrompter.defaultStrategy);
+
+			if (!mapStrategy) {
+				throw new Error('No strategy provided');
+			}
+
+			strategyToRun = new mapStrategy(message, strategyOptions);
+		}
+
+		this.strategy = strategyToRun;
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter An author object to validate or a {@link https://discord.js.org/#/docs/main/stable/typedef/CollectorFilter CollectorFilter} predicate callback.
+	 * @param explicitReturn Wheter or not you wan't an explicit return object
+	 */
+	public run(channel: TextChannel | NewsChannel | DMChannel, authorOrFilter: User | CollectorFilter) {
+		return this.strategy.run(channel, authorOrFilter);
+	}
+
+	/**
+	 * The available strategies
+	 */
+	// @ts-expect-error 2322
```

Comment:
What's 2322? Because this gets removed once compiled and end-users will get errors unless they have `skipLibCheck` enabled (which they sh0updnt need to have)

Reply:
```
No overload matches this call.
  Overload 1 of 3, '(iterable: Iterable<readonly [string, typeof MessagePrompterConfirmStrategy]>): Map<string, typeof MessagePrompterConfirmStrategy>', gave the following error.
    Argument of type '([string, typeof MessagePrompterConfirmStrategy] | [string, typeof MessagePrompterNumberStrategy] | [string, typeof MessagePrompterReactionStrategy] | [...])[]' is not assignable to parameter of type 'Iterable<readonly [string, typeof MessagePrompterConfirmStrategy]>'.
      The types returned by '[Symbol.iterator]().next(...)' are incompatible between these types.
        Type 'IteratorResult<[string, typeof MessagePrompterConfirmStrategy] | [string, typeof MessagePrompterNumberStrategy] | [string, typeof MessagePrompterReactionStrategy] | [...], any>' is not assignable to type 'IteratorResult<readonly [string, typeof MessagePrompterConfirmStrategy], any>'.
          Type 'IteratorYieldResult<[string, typeof MessagePrompterConfirmStrategy] | [string, typeof MessagePrompterNumberStrategy] | [string, typeof MessagePrompterReactionStrategy] | [...]>' is not assignable to type 'IteratorResult<readonly [string, typeof MessagePrompterConfirmStrategy], any>'.
            Type 'IteratorYieldResult<[string, typeof MessagePrompterConfirmStrategy] | [string, typeof MessagePrompterNumberStrategy] | [string, typeof MessagePrompterReactionStrategy] | [...]>' is not assignable to type 'IteratorYieldResult<readonly [string, typeof MessagePrompterConfirmStrategy]>'.
              Type '[string, typeof MessagePrompterConfirmStrategy] | [string, typeof MessagePrompterNumberStrategy] | [string, typeof MessagePrompterReactionStrategy] | [...]' is not assignable to type 'readonly [string, typeof MessagePrompterConfirmStrategy]'.
                Type '[string, typeof MessagePrompterNumberStrategy]' is not assignable to type 'readonly [string, typeof MessagePrompterConfirmStrategy]'.
                  Types of property '1' are incompatible.
                    Type 'typeof MessagePrompterNumberStrategy' is missing the following properties from type 'typeof MessagePrompterConfirmStrategy': confirmEmoji, cancelEmojits(2769)
                    
```

Reply:
No clue what to do with this one. Discussed this with Nytlife & Stitch when it happened